modules/types/include/hephaestus/types/uuid_v4.h

```diff
@@ -0,0 +1,42 @@
+//=================================================================================================
+// Copyright (C) 2023-2024 HEPHAESTUS Contributors
+//=================================================================================================
+
+#pragma once
+
+#include <cstdint>
+#include <random>
+
+namespace heph::types {
+
+/// @brief A class representing a UUID (Universally Unique Identifier) version 4, which is purely (pseudo)
+/// randomly generated. We follow RFC 9562, which defines the UUIDv4 format. For details, see
+/// https://www.rfc-editor.org/rfc/rfc9562.html.
+struct UuidV4 {
+  [[nodiscard]] auto operator==(const UuidV4&) const -> bool = default;
+
+  /// @brief Generates a random valid UUIDv4 using the provided random number generator.
+  [[nodiscard]] static auto random(std::mt19937_64& mt) -> UuidV4;
+
+  /// @brief Creates a UUIDv4 using an internal static random number generator.
+  [[nodiscard]] static auto create() -> UuidV4;
+
+  /// @brief Creates a UUIDv4 with all bits set to zero, 00000000-0000-0000-0000-000000000000.
+  /// A Nil UUID value can be useful to communicate the absence of any other UUID value in situations that
+  /// otherwise require or use a 128-bit UUID. A Nil UUID can express the concept "no such value here". Thus,
+  /// it is reserved for such use as needed for implementation-specific situations.
+  [[nodiscard]] static auto createNil() -> UuidV4;
+
+  /// @brief Creates a UUIDv4 with all bits set to one, FFFFFFFF-FFFF-FFFF-FFFF-FFFFFFFFFFFF.
+  /// A Max UUID value can be used as a sentinel value in situations where a 128-bit UUID is required, but a
+  /// concept such as "end of UUID list" needs to be expressed and is reserved for such use as needed for
+  /// implementation-specific situations.
+  [[nodiscard]] static auto createMax() -> UuidV4;
+
+  [[nodiscard]] auto format() const -> std::string;
+
```

Comment:
We don't have a default ctor here. What should the semantics be? Right now, it's undefined. Options would be: nil, max or a valid v4 UUID. I'm leaning towards nil.